src/core/tests/constant.cpp

```diff
@@ -934,6 +1166,126 @@ TEST(constant, uint4_write_then_cast_custom_type) {
     EXPECT_THAT(v, ElementsAre(1, 3, 12));
 }
 
+//
+// uint6
+//
+
+TEST(constant, uint6_string) {
+    const auto shape = Shape{4};
+
+    op::v0::Constant c(element::u6, shape, vector<string>{"4", "9", "15", "16"});
+    auto v = c.cast_vector<uint8_t>();
+
+    ASSERT_EQ(v.size(), shape_size(shape));
+    EXPECT_THAT(v, ElementsAre(4, 9, 15, 16));
+
+    const auto p = c.get_data_ptr<uint8_t>();
+    EXPECT_EQ(p[0], 0x49);
+    EXPECT_EQ(p[1], 0xf0);
+    EXPECT_EQ(p[2], 0b00000001);
+
+    EXPECT_EQ(c.convert_value_to_string(2), "15");
+    EXPECT_THAT(c.get_value_strings(), ElementsAre("4", "9", "15", "16"));
+}
+
+TEST(constant, uint6_string_broadcast) {
+    const auto shape = Shape{4};
+
+    op::v0::Constant c(element::u6, shape, vector<string>{"5"});
+    auto v = c.cast_vector<uint8_t>();
+
+    ASSERT_EQ(v.size(), shape_size(shape));
+    EXPECT_THAT(v, Each(5));
+
+    const auto p = c.get_data_ptr<uint8_t>();
+    EXPECT_EQ(p[0], 0x55);
+    EXPECT_EQ(p[1], 0x55);
+    EXPECT_EQ(p[2], 0b00000000);
+}
+
+TEST(constant, uint6_vector_less_than_one_storage_unit) {
+    auto const shape = Shape{3};
+    const auto input = std::vector<uint8_t>{5, 23, 1};
+
+    op::v0::Constant c(element::u6, shape, input);
+    auto v = c.cast_vector<uint8_t>();
+
+    ASSERT_EQ(v.size(), shape_size(shape));
+    EXPECT_THAT(v, ElementsAre(5, 23, 1));
+
+    const auto p = c.get_data_ptr<uint8_t>();
+    EXPECT_EQ(p[0], 0x57);
+    EXPECT_EQ(p[1] & 0xF0, 0x10);
+    EXPECT_EQ(p[2] & 0b1111100, 0b00010000);
+}
+
+TEST(constant, uint6_vector_greater_than_one_storage_unit) {
+    auto const shape = Shape{6};
+    const auto input = std::vector<uint8_t>{25, 3, 1, 0, 45, 5};
+
+    op::v0::Constant c(element::u6, shape, input);
+    auto v = c.cast_vector<uint8_t>();
+
+    ASSERT_EQ(v.size(), shape_size(shape));
+    EXPECT_THAT(v, ElementsAre(25, 3, 1, 0, 45, 5));
+
+    const auto p = c.get_data_ptr<uint8_t>();
+    EXPECT_EQ(p[0], 0x93);
+    EXPECT_EQ(p[1], 0x10);
+    EXPECT_EQ(p[2], 0b01000000);
+
+    EXPECT_EQ(p[3], 0xd5);
+    // p[4] ignore
+    EXPECT_EQ(p[5] & 0b11110000, 0b10000000);
+}
+
+TEST(constant, uint6_vector_broadcast) {
+    const auto shape = Shape{4};
+    op::v0::Constant c(element::u6, shape, vector<int8_t>{45});
+
+    auto v = c.cast_vector<uint8_t>();
+    ASSERT_EQ(v.size(), shape_size(shape));
+    EXPECT_THAT(v, Each(45));
+
+    const auto p = c.get_data_ptr<uint8_t>();
+    EXPECT_EQ(p[0], 0xdd);
+    EXPECT_EQ(p[1], 0xdd);
+    EXPECT_EQ(p[2], 0b10101010);
```

Comment:
What these checks mean?

Reply:
The Constant is filled with four values of `45 (x2d)`. It will consume 3 bytes as u6 can store max 4 values on 3 bytes. The u6 pack value in bytes (`4bits (byte0 or byte1) + 2bits (byte2)`)

The bytes 0,1 contains LSB of 45 (0xd). In these two bytes there are four `0xd` values. The byte 2 contains MSB of 45 value (x2) and there is four `0x2` in this byte.